templates/components/form.jsx

```diff
@@ -1,44 +1,58 @@
 var React = require('react/addons');
 var ValidationMixin = require('react-validation-mixin');
-var Joi = require('joi');
 
 var Form = React.createClass({
   propTypes: {
     fields: React.PropTypes.array.isRequired
   },
   statics: {
-      'phoneLabels': {
-        'username': 'phone-label-username',
-        'password': 'phone-label-password'
+      'iconLabels': {
+        'username': 'icon-label-username',
+        'password': 'icon-label-password',
+        'email':    'icon-label-email'
       }
   },
   mixins: [
     ValidationMixin,
     React.addons.LinkedStateMixin
   ],
-  validatorTypes:  {
-    username: Joi.string().min(1).required().label('Username'),
-    password: Joi.string().min(3).required().label('Password')
+  validatorTypes: false,
+  componentWillMount: function() {
+    this.validatorTypes = this.props.validators;
```

Comment:
remove what? `this` gets cleaned up after unmount, so there shouldn't be any lingering data. If these were event handlers: yes. As just plain properties, there should be no reason to.